### PR TITLE
Unescape html escape characters in reddit text and titles

### PR DESCRIPTION
**Pull request by asleep-cult**
Use `html.unescape` to replace html escape characters with their Unicode equivalent in reddit text and titles.
Examples:
Capture
Capture

### CODE REVIEW

File: bot/exts/info/reddit.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import asyncio
 import logging
+import html
```
Comment (Shivansh-007) [inline bot/exts/info/reddit.py:3]: We don’t really need the full library here. 
```suggestion
from html import unescape
```

**Review by Shivansh-007 — CHANGES_REQUESTED** [review]
Nitpick

**Review by Shivansh-007 — APPROVED** [review]
Thanks! Looks Good To Me!

File: bot/exts/info/reddit.py [hunk 2]
```diff
@@ -188,7 +189,8 @@ async def get_top_posts(self, subreddit: Subreddit, time: str = "all", amount: i
             comments = data["num_comments"]
             author = data["author"]
 
-            title = textwrap.shorten(data["title"], width=64, placeholder="...")
+            title = unescape(data["title"])
+            title = textwrap.shorten(title, width=64, placeholder="...")
```
Comment (ChrisLovering) [inline bot/exts/info/reddit.py:193]: Could we have this and the line above in a single assignment?
```suggestion
            title = textwrap.shorten(
                unescape(data["title"]),
                width=64,
                placeholder="..."
            )
```

File: bot/exts/info/reddit.py [hunk 3]
```diff
@@ -179,7 +180,7 @@ async def get_top_posts(self, subreddit: Subreddit, time: str = "all", amount: i
         for post in posts:
             data = post["data"]
 
-            text = data["selftext"]
+            text = unescape(data["selftext"])
             if text:
```
Comment (ChrisLovering) [inline bot/exts/info/reddit.py:184]: The line above could be moved into this if statement as a walrus operator
```suggestion
            if text := unescape(data["selftext"]):
```

**Review by ChrisLovering — CHANGES_REQUESTED** [review]
We usually recommend that you raise issues before raising a PR directly to the repo, as it's possible the core dev team will have opinions.

As far this change, we are currently porting this over to [Sir Lancebot](https://github.com/python-discord/sir-lancebot) with https://github.com/python-discord/sir-lancebot/issues/530 and https://github.com/python-discord/sir-lancebot/pull/569

If we decide that we want these changes in the meantime, my comments are below.

**Comment by Xithrius** [comment]
@asleep-cult Will you be continuing this PR?

**Review by ChrisLovering — APPROVED** [review]

**Review by vcokltfre — APPROVED** [review]

**Comment by Xithrius** [comment]
We'll merge this for the time being, should be fixed in Reddit migration to Sir-Lancebot.

**Pull request merged**